lib/output.js

```diff
@@ -116,6 +116,38 @@ function toBuffer (options, callback) {
   return this._pipeline(is.fn(options) ? options : callback);
 }
 
+/**
+ * Same as toBuffer, but writes output to a Uint8Array.
+ *
+ * @example
+ * sharp(input)
+ *   .toBuffer((err, data, info) => { ... });
+ *
+ * @example
+ * sharp(input)
+ *   .toUint8Array()
+ *   .then(data => { ... })
+ *   .catch(err => { ... });
+ *
+ * @example
+ * sharp(input)
+ *   .toUint8Array({ resolveWithObject: true })
+ *   .then(({ data, info }) => { ... })
+ *   .catch(err => { ... });
+ *
+ * @param {Object} [options]
+ * @param {boolean} [options.resolveWithObject] Resolve the Promise with an Object containing `data` and `info` properties instead of resolving only with `data`.
+ * @param {Function} [callback]
+ * @returns {Promise<Buffer>} - when no callback is provided
+ */
+function toUint8Array (options, callback) {
+  return new Promise((resolve, reject) => {
+    this.toBuffer(options, callback).then((buff) => {
+      resolve(Uint8Array.from(buff));
```

Comment:
I think this will copy the data. Would something like `new Uint8Array(buffer, 0, buffer.length)` to create a view be better here?

Reply:
I think there might be a lack of error handling here - what happens when `toBuffer` rejects?

Reply:
Same as what I said about the input, but I'll look into it and how to useArrayBuffer here.



> I think there might be a lack of error handling here - what happens when `toBuffer` rejects?

Oh snap! Yep, I forgot to add that in my last commit, working on it.